File: archive/sender-v1/structs.go
package main

import (
	"market_replay/aes"
	"net"
)

type Member struct {
	ID       int
	Password string
	Status   string
	Traders  []Trader
}

type Trader struct {
	TID        int
	MID        int
	Password   string
	HostIP     string
	Port       string
	Connected  bool
	CGW_Res    GW_Response
	conn       net.Conn
	msg_seq    int
	CTX        *aes.CipherContext
	AckCount   int
	throttle   uint32
	traderChan chan redisData
	session_id int
}

type GW_Response struct {
	GW1    string
	Port1  uint32
	GW2    string
	Port2  uint32
	SecKey string
	IV     string
}

type DataPacket struct {
	// Message      string
	InstrumentId int
	Price        int
	TraderId     int
	ProductId    int
}

type TimeStamps struct {
	GW_Req_In  uint64
	GW_ME_Req  uint64
	ME_Req_In  uint64
	ME_Res_Out uint64
	GW_ME_Res  uint64
	GW_Res_Out uint64
}

type Record struct {
	InstrumentID string
	LowerLimit   int
	UpperLimit   int
	MinLot       int
	Product_ID   int
}

type ProcessConfig struct {
	TraderCount        int
	ThrottleLimit      int
	RunTimeLimit       int
	TargetEnv          int
	FileType           int
	OrderModPercent    int
	OrderCancelPercent int
	HeartBeat          int
}

type redisData struct {
	TemplateID      int     `json:"template_id"`
	Price           float64 `json:"price"`
	OrderQty        int     `json:"order_qty"`
	MaxShow         int     `json:"maxshow"`
	InstID          int     `json:"instid"`
	AccType         string  `json:"acc_type"`
	TimeInForce     string  `json:"time_in_force"`
	ClientCode      string  `json:"client_code"`
	MsgSeqNum       int     `json:"msg_seq_num"`
	TraderId        int
	PartitionId     int
	ProductId       int
	timeStamp       int64
	TransactionType int
	orderID         float64
}

type CacheEntry struct {
	NewOrderID          int64
	UpdatedActivityTime int64
	Used                bool
}

File: archive/DataCreation-v1/modules/structs.py
class RedisData:
    
    def __init__(self,template_id,price,order_qty,maxshow,instid,acc_type,time_in_force,clientCode,MsgSeqNum,transactionType,OrderNumber,memberID, actualTraderID, TransactionTimeStamp,TradingProductId,PartitionID, trader_id = 0,partition_id = 0,product_id = 0,timestamp = 0):
        self.template_id = template_id
        self.instid = instid
        self.price = price
        self.order_qty = order_qty
        self.maxshow = maxshow
        self.acc_type = acc_type
        self.time_in_force = time_in_force
        self.clientCode = clientCode
        self.MsgSeqNum = MsgSeqNum
        self.trader_id = trader_id
        self.product_id = product_id
        self.timestamp = timestamp
        self.transactionType = transactionType
        self.OrderNumber = OrderNumber
        self.TransactionTimeStamp = TransactionTimeStamp
        self.TradingProductID = TradingProductId
        self.memberID = memberID
        self.actualTraderID = actualTraderID
        self.PartitionID = PartitionID
        self.partition_id = partition_id
        
        

    def to_dict(self):
        return {
            "template_id": self.template_id,
            "instid": self.instid,
            "price": self.price,
            "order_qty": self.order_qty,
            "maxshow": self.maxshow,
            "acc_type": self.acc_type,
            "time_in_force": self.time_in_force,
            "clientCode": self.clientCode,
            "MsgSeqNum" : self.MsgSeqNum,
            "transactionType":self.transactionType,
            "OrderNumber":self.OrderNumber,
            "TransactionTimeStamp": self.TransactionTimeStamp,
            "TradingProductID": self.TradingProductID,
            "member_id": self.memberID,
            "trader_id": self.actualTraderID,
            "PartitionID": self.PartitionID
        }
    
    def toCommaSeparated(self):
        return ",".join([
            str(self.template_id),
            str(self.instid),
            str(self.price),
            str(self.order_qty),
            str(self.maxshow),
            str(self.acc_type),
            str(self.time_in_force),
            str(self.clientCode),
            str(self.MsgSeqNum),
            str(self.transactionType),
            str(self.OrderNumber),
            str(self.TransactionTimeStamp),
            str(self.TradingProductID),
            str(self.memberID),
            str(self.actualTraderID),
            str(self.PartitionID)
        ])



class Trader:
    def __init__(self, tid, host_ip, port, mid, password):
        self.tid = tid
        self.host_ip = host_ip
        self.port = port
        self.mid = mid
        self.password = password

    def to_dict(self):
        return {
            "TID": self.tid,
            "HostIP": self.host_ip,
            "Port": self.port,
            "MID": self.mid,
            "Password": self.password,
        }

    @staticmethod
    def from_dict(data):
        return Trader(
            tid=data["TID"],
            host_ip=data["HostIP"],
            port=data["Port"],
            mid=data["MID"],
            password=data["Password"],
        )
